api/lib/ZwaveClient.ts

```diff
@@ -2483,10 +2466,54 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 
 	// ------------NODES MANAGEMENT-----------------------------------
 
+	async getStoreNodes() {
+		if (!this.homeHex) {
+			throw new Error('HomeHex not set')
+		}
+
+		let nodes = jsonStore.get(store.nodes)
+
+		// back compatibility fixes
+
+		// convert store nodes from array to object
+		if (Array.isArray(nodes)) {
+			const storeNodes = {}
+
+			for (let i = 0; i < nodes.length; i++) {
+				if (nodes[i]) {
+					storeNodes[i] = nodes[i]
+				}
+			}
+
+			nodes = storeNodes
+		}
+
+		const keys = Object.keys(nodes)
+
+		// ensure store nodes are stored using homeHex
+		if (keys.length > 0 && !keys[0].startsWith('0x')) {
+			this.storeNodes = nodes
+			await jsonStore.put(store.nodes, {
+				[this.homeHex]: nodes,
+			})
+		} else {
```

Comment:
this is the back compatibility trick I added. Basically I just check if the keys in nodes.json are strings starting with 0x (homeHex), if not it means them are not converted to the new format yet so I store the keys in the new format

Reply:
To keep this simply and less triky I decided to change the format of nodes.json instead of changing the filename itself as that would have required a rewrite of the jsonstore that I prefer to avoid